odxtools/cli/_print_utils.py

```diff
@@ -243,8 +243,11 @@ def print_dl_metrics(variants: List[DiagLayer], print_fn: Callable[..., Any] = p
         name.append(variant.short_name)
         type.append(variant.variant_type.value)
         num_services.append(len(all_services))
-        num_dops.append(len(variant.diag_data_dictionary_spec.data_object_props))
-        num_comparam_refs.append(len(variant.comparam_refs))
+        if (ddds := variant.diag_data_dictionary_spec) is not None:
+            num_dops.append(len(ddds.data_object_props))
+        else:
+            num_dops.append(0)
+        num_comparam_refs.append(len(getattr(variant, "comparams_refs", [])))
```

Comment:
What's going on here?

Reply:
not all diag layers provide `.comparam_refs` anymore. (Also, the `compare` tool is in serious need of refactoring...)